### PR TITLE
Rename filter variables and methods in DataApiRequest

### PR DESCRIPTION
**Pull request by mpardesh**
Addresses #497

### CODE REVIEW

**Comment by mpardesh** [comment]
Looks like I accidentally mixed some commits here...I will fix this

**Comment by archolewa** [comment]
Note that this is a breaking change (anybody who uses either of the getters in a request mapper will experience compile errors when they upgrade).

Instead, we should add new methods with the desired names, and deprecate the old methods.

Also we should add a CHANGELOG.

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java [hunk 1]
```diff
@@ -1009,8 +1009,13 @@ public Granularity getGranularity() {
         return this.intervals;
     }
 
+    public Map<Dimension, Set<ApiFilter>> getApiFilters() {
+        return this.apiFilters;
+    }
+
+    @Deprecated
     public Map<Dimension, Set<ApiFilter>> getFilters() {
```
Comment (archolewa) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java]: These deprecated methods should delegate to the non-deprecated version (i.e. `getFilters()` should return `getApiFilters()`. Furthermore, we should add a Javadoc, with a `@deprecated` annotation indicating which method to use instead.

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java [hunk 2]
```diff
@@ -1020,9 +1025,28 @@ public Granularity getGranularity() {
...
+     * <p>
+     * The Druid filters are built (an expensive operation) every time this method is called. Use it judiciously.
+     *
+     * @return the Druid filter
+     *
+     * @deprecated this method has been deprecated because its name is unclear. Please use {@link #getDruidFilter()}.
+     */
+    @Deprecated
     public Filter getFilter() {
         try (TimedPhase timer = RequestLog.startTiming("BuildingDruidFilter")) {
```
Comment (archolewa) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java]: This should just delegate to `getDruidFilter`.

**Comment by archolewa** [comment]
@mpardesh Just reviewed this change. Had a few small comments. Once those are addressed, I'll be willing to give my approval.

**Comment by michael-mclawhorn** [comment]
There's a problem with the underlying issue.

There's a reason that API filters are considered plural while Druid filters are singular.  Because API filters are treated as a list (or a list grouped into a map by dimension) while the Druid filters are a tree with a leaf node as the reference variable.

**Comment by archolewa** [comment]
@michael-mclawhorn As far as I can tell, the api filters method is still called `getApiFilters`, while the druid filter is called `getDruidFilter`. So things are still singular and plural.

**Comment by michael-mclawhorn** [comment]
@archolewa Yeah, I see that.
I actually think I'd rather make a breaking change than each the deprecation cost here, but I suppose that's not the friendliest choice.  This is going to create rebase pain with https://github.com/yahoo/fili/pull/501, so I'd merge it quickly if you don't want to fix it.

**Pull request merged**